pacman/model/graphs/machine/machine_fpga_vertex.py

```diff
@@ -75,5 +76,12 @@ def virtual_chip_y(self):
 
     @overrides(AbstractVirtual.set_virtual_chip_coordinates)
     def set_virtual_chip_coordinates(self, virtual_chip_x, virtual_chip_y):
-        self._virtual_chip_x = virtual_chip_x
-        self._virtual_chip_y = virtual_chip_y
+        if virtual_chip_x is not None and virtual_chip_y is not None:
+            self._virtual_chip_x = virtual_chip_x
+            self._virtual_chip_y = virtual_chip_y
+            self.add_constraint(ChipAndCoreConstraint(
+                self._virtual_chip_x, self._virtual_chip_y))
+            if (self._app_vertex is not None and
```

Comment:
This assumes that a virtual application vertex only has one X,Y location. That seems wrong-headed.